realm/realm-annotations-processor/src/main/java/io/realm/processor/Backlink.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.processor;
+
+import java.util.List;
+
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+import io.realm.annotations.LinkingObjects;
+import io.realm.annotations.Required;
+
+/**
+ * A Backlink is:
+ * <p>
+ * <code>
+ * class TargetClass {
+ *     // ...
+ *     {@literal @}LinkingObjects("sourceField")
+ *     RealmResults&lt;SourceClass&gt; targetField;
+ * }
+ * </code>.
+ * <p>
+ * The targetField of a managed object cannot be assigned.  It can be queried normally.
+ * When an instance X of a class with such a declaration is copied to Realm (`copyToRealm()`),
```

Comment:
I think it means that if you create a RealmObject like so

``` java
public class MyObject extends RealmObject {
    @LinkingObjects(/* not sure what goes here */)
    RealmResults<OtherObject> otherObjects;

    public void setOtherObjects(RealmResults<OtherObject> otherObjects) {
        this.otherObjects = otherObjects;
    }
}
```

Then:

- in case of managed object, this setter is invalid, because you cannot modify a Backlink of the linking objects

- in case of unmanaged object, if you do `copyToRealm()` with the `RealmResults` field set, then this RealmResults field is completely ignored and replaced with the actual backlink.